hax/hax/motr/__init__.py

```diff
@@ -282,6 +282,12 @@ def ha_obj_state(st):
                 notify_devices = False
             notes += self._generate_sub_services(note, self.consul_util,
                                                  notify_devices)
+            if (st.fid.container == ObjT.PROCESS.value
+                    and st.status == ServiceHealth.OK):
+                if self.consul_util.is_ios_service(st.fid):
```

Comment:
I don't think you need to check if its an ioservice or some other. You just need to get the drives corresponding to that service and set the status of that drive to ONLINE in this case.

Reply:
Probably this function is getting too overloaded, may be having a function map would help here. We can do that as improvisation though.

Reply:
> I don't think you need to check if its an ioservice or some other. You just need to get the drives corresponding to that service and set the status of that drive to ONLINE in this case.

yes correct, done